### ISSUE TITLE
AIX 7.1 - Attempts to install says I must run make, despite already running make.

### ISSUE THREAD

**Issue by Teej42** [xianyi/OpenBLAS#1896]
I am still using the GA version of 0.3.3, with modification to memory.c, Makefile.install, and Makefile.system.  I am getting this message:

```
$ sudo make install
The license for the Evaluation version of IBM XL C/C++ for AIX V11.1 compiler product has expired. Please send an email to compiler@ca.ibm.com for information on purchasing the product.
make: *** [Makefile.prebuild:62: getarch] Error 1
make -j 1 -f Makefile.install install
make[1]: Entering directory '/scratch/focal_point/Packages/OpenBLAS'
Makefile.install:21: *** OpenBLAS: Please run "make" firstly.  Stop.
make[1]: Leaving directory '/scratch/focal_point/Packages/OpenBLAS'
make: *** [Makefile:336: install] Error 2
```

Licensing message disappears if I use the same value I use for the compile, which to remind you is:

`make BINARY=64 CC=gcc FC=gfortran FFLAGS="-O2 -maix64" RANLIB=ranlib LD=ld AR=ar MAKE_NB_JOBS=1`

Same error message.  This is my 5th AIX 7.1 server install, and I never saw this one before.  Details attached for the make -d install:

[log.txt](https://github.com/xianyi/OpenBLAS/files/2634484/log.txt)

I don't understand why there's an error with getarch.

I am cleaning and recompiling and will update this ticket accordingly with the logs.

**Comment by brada4** [comment]
Your "compiler" did not compile "getarch" detection program.
You need a C compiler that emits binaries, like fine free gcc.

**Comment by Teej42** [comment]
To clarify this - I did invoke the compiler in the other make:

`make BINARY=64 CC=gcc FC=gfortran FFLAGS="-O2 -maix64" RANLIB=ranlib LD=ld AR=ar MAKE_NB_JOBS=1`

[make.txt](https://github.com/xianyi/OpenBLAS/files/2634676/make.txt)

`sudo make -d CC=gcc install > install.txt 2>&1`

[install.txt](https://github.com/xianyi/OpenBLAS/files/2634682/install.txt)

As you can see, even when invoking gcc as the compiler, it continues to demand that I run make first, despite already doing so immediately before.

Another reminder:  This successfully worked on other AIX platforms, and I confirmed that the makefile changes matches on from the working environments and this one.  What can I do to expand on the reason for `make install` to demand I run `make` again?

To help, here's the modified Makefiles:

[Makefile.install.txt](https://github.com/xianyi/OpenBLAS/files/2634691/Makefile.install.txt)
[Makefile.system.txt](https://github.com/xianyi/OpenBLAS/files/2634692/Makefile.system.txt)

**Comment by martin-frbg** [comment]
Does it work when you invoke the `make install` with both `CC=gcc` and `BINARY=64` ?

**Comment by Teej42** [comment]
Unfortunately, no.

```
$ sudo make BINARY=64 CC=gcc install
Password:
make -j 2 -f Makefile.install install
make[1]: Entering directory '/scratch/focal_point/Packages/OpenBLAS'
Makefile.install:21: *** OpenBLAS: Please run "make" firstly.  Stop.
make[1]: Leaving directory '/scratch/focal_point/Packages/OpenBLAS'
make: *** [Makefile:336: install] Error 2
```

**Comment by martin-frbg** [comment]
Strange. I see now that the Makefile.install is acting on a simple token - it expects an empty file "lib.grd" to be present as a sign that `make` has been run before (there is a `touch lib.grd` at the end of the `libs` target in Makefile). Do you have this file in /scratch/focal_point/Packages/OpenBLAS ?

**Comment by Teej42** [comment]
Yup:

```
$ ls -al lib.grd
-rw-r--r--    1 Me  staff             0 Nov 30 13:13 lib.grd
```

**Comment by martin-frbg** [comment]
Just for fun - do you still see it when you do `sudo ls -al lib.grd` ?

**Comment by Teej42** [comment]
```
$ sudo ls -al lib.grd
Password:
-rw-r--r-- 1 Me staff 0 Nov 30 13:13 lib.grd
```

**Comment by martin-frbg** [comment]
Hmm. System time or filesystem timestamp problems perhaps (lib.grd seen to be older than Makefile.install or some such) ?

**Comment by Teej42** [comment]
Sorry, had to take care of things over the weekend:

```
-rw-r--r-- 1 u071625 staff 9355 Nov 30 10:31 Makefile.install
-rw-r--r-- 1 u071625 staff 0 Nov 30 13:13 lib.grd
```

Full file listing:

[result.txt](https://github.com/xianyi/OpenBLAS/files/2639615/result.txt)

**Comment by martin-frbg** [comment]
Strange. Do I understand correctly that the same `make install` worked on a number of other AIX machines, and just one system is giving you this problem ?

**Comment by Teej42** [comment]
That is correct.  The modified files (driver/others/memory.c, Makefile.install, and Makefile.system) was explicitly copied over from a different AIX server to this one.  Also confirmed (oslevel -s) that everything matches (7100-05-01-1731).  The only difference is AIX Toolbox's info-6.4-1.aix6.1.ppc.rpm and rpm.rte 4.13.0.5 on newer server, and info-6.3-1.aix6.1.ppc.rpm and rpm.rte 4.13.0.4 on older server.  I doubt they play a part, but have to mention.

I could chalk it up to "there is a ghost in this server" and package up the existing binaries on a different AIX server, and install on the new server.  Is there a documentation where everything is parked on, or do I have to go through the Makefile.install script to find them all?

**Comment by martin-frbg** [comment]
Guess you could try `touch lib.grd` (or change the keyword `error` in Makefile.install to `info`)... ghost in this server looks plausible if there were no issues with the build as such.

**Comment by Teej42** [comment]
Interesting.  When I changed to `info`, I got this:
```
Makefile.install:24: *** missing separator (did you mean TAB instead of 8 spaces?).  Stop.
```

It turns out that the file had all of its tabs converted to whitespace for some odd reason.  I fixed this, and the install script works.  

So there's a ghost in the sftp software or something else that broke this the first time.  Maybe I watered the file after midnight by mistake... 

Sorry about this, closing.
